select.go

```diff
@@ -117,20 +121,62 @@ func (s *Select) innerRun(starting int, top rune) (int, string, error) {
 	rl.Write([]byte(hideCursor))
 	sb := screenbuf.New(rl)
 
-	rl.Operation.ExitVimInsertMode() // Never use insert mode for selects
+	var searchInput []rune
+	canSearch := s.Searcher != nil
+	searchMode := false
 
 	c.SetListener(func(line []rune, pos int, key rune) ([]rune, int, bool) {
-		switch key {
-		case readline.CharEnter:
+		switch {
+		case key == readline.CharEnter:
 			return nil, 0, true
-		case readline.CharNext, 'j':
+		case key == readline.CharNext || (key == 'j' && !searchMode):
 			s.list.Next()
-		case readline.CharPrev, 'k':
+		case key == readline.CharPrev || (key == 'k' && !searchMode):
 			s.list.Prev()
-		case ' ': // space press
-			s.list.PageDown()
-		case 'b':
+		case key == '/':
+			if !canSearch {
+				break
+			}
+
+			if searchMode {
+				searchMode = false
+				searchInput = nil
+				s.list.CancelSearch()
+			} else {
+				searchMode = true
+			}
+		case key == readline.CharBackspace:
+			if !canSearch || !searchMode {
+				break
+			}
+
+			if len(searchInput) > 1 {
+				searchInput = searchInput[:len(searchInput)-1]
+				s.list.Search(string(searchInput))
+			} else {
+				searchInput = nil
+				s.list.CancelSearch()
+			}
+		case key == readline.CharBackward || (key == 'h' && !searchMode):
 			s.list.PageUp()
+		case key == readline.CharForward || (key == 'l' && !searchMode):
+			s.list.PageDown()
+		default:
+			if canSearch && searchMode {
+				searchInput = append(searchInput, line...)
+				s.list.Search(string(searchInput))
+			}
+		}
+
+		if searchMode {
+			header := fmt.Sprintf("Search: %s%s", string(searchInput), cursor)
+			sb.WriteString(header)
+		} else {
+			header := "Use the arrow keys to navigate: ↑↓←→"
```

Comment:
We should space the characters out so that it flows better with the `and / for search`.

`Use the arrow keys to navigate: ↑ ↓ ← → and / for search`

Ultimately I think we should come back and make the headers toggleable, to facilitate a more custom output.